kartothek/io/dask/dataframe.py

```diff
@@ -606,3 +604,67 @@ def hash_dataset(
             .apply(_unpack_hash, unpack_meta=ddf._meta, subset=subset, meta="uint64")
             .astype("uint64")
         )
+
+
+def calculate_repartition_ratio(
+    dataset_factory: DatasetFactory,
+    predicates: Optional[PredicatesType] = None,
+    *,
+    target_nrows_per_partition: int,
+    dataset_metadata_frac: float,
+) -> dd.core.Scalar:
+    """
+    Calculate a `repartition_ratio` according to the dataset input and `target_nrows_per_partition` (the desired
+    number of rows of each dataframe partition).
+
+
```

Comment:
1. please add a note that this function performs IO.
2. "calculate" could be read by many users as "compute on client". I would prefer something like `compute`.

Reply:
`calculate_repartion_ratio()` function is removed.